src/clj_deconstruct_sigs/core.clj

```diff
@@ -119,12 +119,13 @@
         used-indices (map first pruned)
         seed-idx (find-seed sample-tumor sigs)
         w (loop [w (m/set-column (m/zero-array [1 (count sigs)]) seed-idx 10)
-                 error-diff Double/POSITIVE_INFINITY]
```

Comment:
I just  remembered I took this from the original impl, but not sure why.

https://github.com/raerose01/deconstructSigs/blob/ca3c2f9a22a7d12944684c082d65b4d5f9a478ac/R/whichSignatures.R#L141

Reply:
Sorry. It's my mistake. I have changed when working to prevent "Auto-boxing" warning. But it has no effect.

Reply:
Probably this should fix the auto-boxing warning?:

```diff
         w (loop [w (m/set-column (m/zero-array [1 (count sigs)]) seed-idx 10)
                  error-diff Double/POSITIVE_INFINITY]
             (if (< 1e-3 error-diff)
-              (let [error-pre ^double (get-error sample-tumor sigs w)
+              (let [error-pre (double (get-error sample-tumor sigs w))
                     new-w (update-W-GR sample-tumor sigs w)
-                    error-post ^double (get-error sample-tumor sigs new-w)]
+                    error-post (double (get-error sample-tumor sigs new-w))]
                 (recur new-w (/ (- error-pre error-post) error-pre)))
               w))
```

Reply:
It works. Just fixed in https://github.com/chrovis/clj-deconstruct-sigs/commit/312725d126e11831a036c867151505ff62d75f2c. Thank you!